src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs

```diff
@@ -229,5 +231,63 @@ internal static void ValidateNumber(ReadOnlySpan<byte> utf8FormattedNumber)
                     nameof(utf8FormattedNumber));
             }
         }
+
+        private static readonly UTF8Encoding s_utf8Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+
+        public static unsafe bool IsValidUtf8String(ReadOnlySpan<byte> bytes)
+        {
+            try
+            {
+#if NETCOREAPP
+                s_utf8Encoding.GetCharCount(bytes);
```

Comment:
@GrabYourPitchforks, I thought we had a Utf8.IsValid API, but I can't find it now in either source or an issue.  Am I making it up?  Is that something we plan to expose?

Reply:
Looks like roslyn is also using similar method for `CS9026` diagnostics in `""u8` strings: https://github.com/dotnet/roslyn/blob/ab7dd82/src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs#L113

`IsValid(ReadOnly<byte>)` is a good candidate API for `System.Text.Unicode.Utf8` type. :)